trades.py

```diff
@@ -162,12 +162,12 @@ def __repr__(self) -> str:
         return self.name
 
     @staticmethod
-    def argparse(s):
-        """Performs string conversion to Attribute."""
+    def argparse(s: str) -> "MarketAttribute":
+        """Performs string conversion to MarketAttribute."""
         try:
             return MarketAttribute[s.upper()]
         except KeyError:
-            return s
+            return MarketAttribute(s)
```

Comment:
Can you explain? Will this not raise the exception again?

Reply:
Changed to raise a ValueError, which is handled by the argparser.